src/miner.cpp

```diff
@@ -526,6 +527,10 @@ void static BitcoinMiner(const CChainParams& chainparams)
                     // Changing pblock->nTime can change work required on testnet:
                     hashTarget.SetCompact(pblock->nBits);
                 }
+
+                /* If we updated the block above, we also have to update
+                   the auxpow object to match the new block hash.  */
+                CAuxPow::initAuxPow(*pblock);
```

Comment:
`UpdateTime` may change the block's `nTime` and on testnet even `pblock->nBits`.